src/cmd/linuxkit/run_vbox.go

```diff
@@ -0,0 +1,266 @@
+	}
+
+	// vbox options
+	vboxmanageFlag := flags.String("vboxmanage", "VBoxManage", "VBoxManage binary to use")
+	keep := flags.Bool("keep", false, "Keep the VM after finishing")
+	vmName := flags.String("name", "", "Name of the Virtualbox VM")
+	state := flags.String("state", "", "Path to directory to keep VM state in")
+
+	// Paths and settings for disks
+	var disks Disks
+	flags.Var(&disks, "disk", "Disk config, may be repeated. [file=]path[,size=1G][,format=raw]")
+
+	// VM configuration
+	cpus := flags.String("cpus", "1", "Number of CPUs")
+	mem := flags.String("mem", "1024", "Amount of memory in MB")
+
+	// booting config
+	isoBoot := flags.Bool("iso", false, "Boot image is an ISO")
+	uefiBoot := flags.Bool("uefi", false, "Use UEFI boot")
+
+	// networking
+	networking := flags.String("networking", "nat", "Networking mode. null|nat|bridged|intnet|hostonly|generic|natnetwork[<devicename>]")
+
+	if err := flags.Parse(args); err != nil {
+		log.Fatal("Unable to parse args")
+	}
+	remArgs := flags.Args()
+
+	if len(remArgs) == 0 {
+		fmt.Println("Please specify the path to the image to boot")
+		flags.Usage()
+		os.Exit(1)
+	}
+	path := remArgs[0]
+
+	if strings.HasSuffix(path, ".iso") {
+		*isoBoot = true
+	}
+
+	vboxmanage, err := exec.LookPath(*vboxmanageFlag)
+	if err != nil {
+		log.Fatalf("Cannot find management binary %s: %v", *vboxmanageFlag, err)
+	}
+
+	name := *vmName
+	if name == "" {
+		name = strings.TrimSuffix(filepath.Base(path), filepath.Ext(path))
+	}
+
+	if *state == "" {
+		prefix := strings.TrimSuffix(path, filepath.Ext(path))
+		*state = prefix + "-state"
+	}
+	if err := os.MkdirAll(*state, 0755); err != nil {
+		log.Fatalf("Could not create state directory: %v", err)
+	}
+
+	// remove machine in case it already exists
+	cleanup(vboxmanage, name, false)
+
+	_, out, err := manage(vboxmanage, "createvm", "--name", name, "--register")
+	if err != nil {
+		log.Fatalf("createvm error: %v\n%s", err, out)
+	}
+
+	_, out, err = manage(vboxmanage, "modifyvm", name, "--acpi", "on")
+	if err != nil {
+		log.Fatalf("modifyvm --acpi error: %v\n%s", err, out)
+	}
+
+	_, out, err = manage(vboxmanage, "modifyvm", name, "--memory", *mem)
+	if err != nil {
+		log.Fatalf("modifyvm --memory error: %v\n%s", err, out)
+	}
+
+	_, out, err = manage(vboxmanage, "modifyvm", name, "--cpus", *cpus)
+	if err != nil {
+		log.Fatalf("modifyvm --cpus error: %v\n%s", err, out)
+	}
+
+	firmware := "bios"
+	if *uefiBoot {
+		firmware = "efi"
+	}
+	_, out, err = manage(vboxmanage, "modifyvm", name, "--firmware", firmware)
+	if err != nil {
+		log.Fatalf("modifyvm --firmware error: %v\n%s", err, out)
+	}
+
+	// set up serial console
+	_, out, err = manage(vboxmanage, "modifyvm", name, "--uart1", "0x3F8", "4")
+	if err != nil {
+		log.Fatalf("modifyvm --uart error: %v\n%s", err, out)
+	}
+
+	var consolePath string
+	if runtime.GOOS == "windows" {
+		// TODO use a named pipe on Windows
+		log.Fatalf("TODO: add named pipe support for Windows")
```

Comment:
Can we move this earlier? Because otherwise we already create a VM on windows and then error out leaving it around...also the doc should probably not yet mentioned Windows

Reply:
ok, moved